fixql-core/src/main/scala/com/iterable/graphql/compiler/Resolver.scala

```diff
@@ -0,0 +1,23 @@
+package com.iterable.graphql.compiler
+
+import play.api.libs.json.JsObject
+import slick.dbio.DBIO
+
+/** A trait with case class subclasses lets us optimize later */
+trait Resolver[+A] {
+  def jsonFieldName: String
```

Comment:
I thought it would be useful for the Resolver to keep around the name of the field. This turned out to be very inconvenient in the code. So in a later pass I'm going to remove this.

I need something like this to support aliases, so I'll likely move it outside of the Field itself: `subfields: Seq[(String, A)]`